app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java

```diff
@@ -584,15 +650,29 @@ protected void onPostExecute(Void aVoid) {
                 double[] xaxis = digitalChannel.getXAxis();
                 double[] yaxis = digitalChannel.getYAxis();
 
-                String str = "";
+                String str1 = "";
+                String str2 = "";
                 for (int i = 0; i < xaxis.length; i++) {
-                    str += String.valueOf(xaxis[i]);
-                    str += " ";
+                    str1 += String.valueOf(xaxis[i]);
+                    str2 += String.valueOf(yaxis[i]);
```

Comment:
Can you have a look at Java's `StringBuilder` class? It's specially designed for String concatenation.